File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Model/UDS/Implementation/UDSRepositoryModelMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.UDS;
using VecompSoftware.DocSuiteWeb.Model.Entities.UDS;

namespace VecompSoftware.DocSuiteWeb.Mapper.Model.UDS
{
    public class UDSRepositoryModelMapper : BaseModelMapper<UDSRepository, UDSRepositoryModel>, IUDSRepositoryModelMapper
    {

        #region [ Fields ]

        private readonly IUDSSchemaRepositoryModelMapper _udsSchemaRepositoryModelMapper;

        #endregion


        #region [ Constructor ]

        public UDSRepositoryModelMapper(IUDSSchemaRepositoryModelMapper udsSchemaRepositoryModelMapper)
        {
            _udsSchemaRepositoryModelMapper = udsSchemaRepositoryModelMapper;
        }

        #endregion

        #region [ Methods ]

        public static DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus StatusConverter(Entity.UDS.UDSRepositoryStatus status)
        {
            switch (status)
            {
                case Entity.UDS.UDSRepositoryStatus.Draft:
                    return DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus.Draft;
                case Entity.UDS.UDSRepositoryStatus.Confirmed:
                    return DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus.Confirmed;
                default:
                    return DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus.Draft;
            }
        }
        public static Entity.UDS.UDSRepositoryStatus StatusConverter(DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus? status)
        {
            if (!status.HasValue)
            {
                return Entity.UDS.UDSRepositoryStatus.Draft;
            }

            switch (status.Value)
            {
                case DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus.Draft:
                    return Entity.UDS.UDSRepositoryStatus.Draft;
                case DocSuiteWeb.Model.Entities.UDS.UDSRepositoryStatus.Confirmed:
                    return Entity.UDS.UDSRepositoryStatus.Confirmed;
                default:
                    return Entity.UDS.UDSRepositoryStatus.Draft;
            }
        }

        public override UDSRepositoryModel Map(UDSRepository entity, UDSRepositoryModel entityTransformed)
        {
            #region [ Base ]

            entityTransformed.ActiveDate = entity.ActiveDate;
            entityTransformed.ExpiredDate = entity.ExpiredDate;
            entityTransformed.Id = entity.UniqueId;
            entityTransformed.ModuleXML = entity.ModuleXML;
            entityTransformed.Name = entity.Name;
            entityTransformed.SequenceCurrentYear = entity.SequenceCurrentYear;
            entityTransformed.SequenceCurrentNumber = entity.SequenceCurrentNumber;
            entityTransformed.Status = StatusConverter(entity.Status);
            entityTransformed.Version = entity.Version;
            entityTransformed.Alias = entity.Alias;
            entityTransformed.DSWEnvironment = entity.DSWEnvironment;

            #endregion

            #region [ Navigation Properties ]
            entityTransformed.SchemaRepository = entity.SchemaRepository != null ? _udsSchemaRepositoryModelMapper.Map(entity.SchemaRepository, new UDSSchemaRepositoryModel()) : null;
            #endregion

            return entityTransformed;
        }

        #endregion

    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Mappers/UDS/UDSRepositoryModelMapper.ts
import UDSRepositoryModel = require('App/Models/UDS/UDSRepositoryModel');
import BaseMapper = require('App/Mappers/BaseMapper');
import ContainerModelMapper = require('App/Mappers/Commons/ContainerModelMapper');

class UDSRepositoryModelMapper extends BaseMapper<UDSRepositoryModel>{
    constructor() {      
        super();
    }

    public Map(source: any): UDSRepositoryModel {
        let toMap: UDSRepositoryModel = <UDSRepositoryModel>{};

        if (!source) {
            return null;
        }

        toMap.UniqueId = source.UniqueId;   
        toMap.Name = source.Name;
        toMap.SequenceCurrentYear = source.SequenceCurrentYear;
        toMap.SequenceCurrentNumber = source.SequenceCurrentNumber;
        toMap.ModuleXML = source.ModuleXML;
        toMap.Version = source.Version;
        toMap.RegistrationDate = source.RegistrationDate;
        toMap.ActiveDate = source.ActiveDate;
        toMap.ExpiredDate = source.ExpiredDate;
        toMap.DSWEnvironment = source.DSWEnvironment;
        toMap.Alias = source.Alias;
        toMap.Status = source.Status;
        toMap.Container = source.Container ? new ContainerModelMapper().Map(source.Container):null;

        return toMap;
    }
}

export = UDSRepositoryModelMapper;